File: cpp/ros_pubsub_demo/pub.cc
/*
* Copyright (C) Trunk Technology, Inc. - All Rights Reserved
*
* Unauthorized copying of this file, via any medium is strictly prohibited
* Proprietary and confidential
*
* Written by Huang Minhang <huangminhang@trunk.tech>, 2021/11/23 15:02
*/
#include <iostream>
#include <ros/ros.h>
#include <std_msgs/String.h>


int main(int argc, char **argv) {
    std::cout << "Hello, ROS Pub!" << std::endl;
    ros::init(argc, argv, "talker");
    ros::NodeHandle n;
    ros::Publisher pub = n.advertise<std_msgs::String>("chatter", 1);
    ros::Rate loop_rate(10);
    int count = 0;
    while (ros::ok()) {
        std_msgs::String msg;
        std::stringstream ss;
        ss << "hello world " << count;
        msg.data = ss.str();
        pub.publish(msg);
        loop_rate.sleep();
        ++count;
    }
    return 0;
}

File: python/ros_pubsub_demo/pub.py
#!/usr/bin/python3
# -*- coding:UTF-8 -*-
import rospy
from std_msgs.msg import String


if __name__ == '__main__':

    print("pub")
    rospy.init_node('test_talker_node', anonymous=True)
    pub1 = rospy.Publisher('/chatter', String, queue_size=1)
    pub2 = rospy.Publisher('/chatter2', String, queue_size=1)
    pub3 = rospy.Publisher('/chatter3', String, queue_size=1)
    rate = rospy.Rate(10)
    while not rospy.is_shutdown():
        hello_str = "hello world %s" % rospy.get_time()
        pub1.publish(hello_str)
        pub2.publish(hello_str)
        pub3.publish(hello_str)
        rate.sleep()
